File: LunchApp/Data/Models/MenuItem.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Data.Models
{
    public class MenuItem
    {
        public int MenuItemId { get; set; }
        public int MenuSectionId { get; set; }

        [ForeignKey("MenuSectionId")]
        public MenuSection MenuSection { get; set; }

        public string Name { get; set; }
        public int Number { get; set; }
        public int MenuId { get; set; }

        [ForeignKey("MenuId")]
        public Menu Menu { get; set; }
    }
}

File: LunchApp/lunch-app/src/app/models/menu/menu-item.ts
export class MenuItem {
    constructor(menuId: number, menuSectionId: number) {
        this.menuId = menuId;
        this.menuSectionId = menuSectionId;
        this.menuItemId = 0;
        this.number = 1;
        this.name = '';
    }

    number: number;
    name: string;
    menuItemId: number;
    menuSectionId: number;
    menuId: number;
}
